tests/unit/rbtree/main.c

```diff
@@ -245,9 +253,127 @@ void test_rbtree_spam(void)
 	} while (size < MAX_NODES);
 }
 
+/**
+ * @brief Test whether rbtree node struct is embedded
+ * in any user struct
+ *
+ * @details
+ * Initialize a user defined struct contains rbtree node
+ * Add into a rbtree
+ * Enumerate the rbtree node
+ *
+ * @ingroup lib_rbtree_tests
+ *
+ * @see RB_FOR_EACH(),RB_FOR_EACH_CONTAINER()
+ */
+void test_rbtree_container(void)
+{
+	int count = 0;
+	static struct rbtree test_tree_l;
+	struct container_node tree_node[10];
+	struct container_node *c_foreach_node;
+	struct rbnode *foreach_node;
+
+	memset(&test_tree_l, 0, sizeof(test_tree_l));
+	test_tree_l.lessthan_fn = node_lessthan;
+	memset(tree_node, 0, sizeof(tree_node));
+	for (unsigned int i = 0; i < 10; i++) {
+		tree_node[i].value = i;
+		rb_insert(&test_tree_l, &tree_node[i].node);
+	}
+
+	/*"for each" style iteration to verify node*/
+	RB_FOR_EACH(&test_tree_l, foreach_node) {
+		zassert_true(((struct container_node *)foreach_node)->value
+				== count, "RB_FOR_EACH failed");
+		count++;
+	}
+
+	count = 0;
+	/*"for each" style iteration to verify node*/
+	RB_FOR_EACH_CONTAINER(&test_tree_l, c_foreach_node, node) {
+		zassert_true(c_foreach_node->value == count,
+				"RB_FOR_EACH_CONTAINER failed");
+		count++;
+	}
+}
+
+
+
+/**
+ * @brief Test rbtree some operations in logarithmic time
+ * complex
+ *
```

Comment:
FWIW: there's actually validation already in the tree code itself that checks whether or not the maximum tree height is exceeded (because if this didn't work, then the stack-based array created to hold the backpointers during a walk would explode!).

Not really a reason not to add this test, it's cheap.  But this condition is already covered as I understand it.

Reply:
Hi @andyross , thanks for your comments, this condition actually is covered by existing testcase you mentioned, so I have removed it. 

And my test point is some operations running in logarithmic time, red/black tree is a type of data structure without depending on any platform, you know, this test is a unit test,which is separated from all platforms, and I use the APIs in usr/libc of native posix to get time of code running instead of methods before. I think it's more intuitively to test the point. 

And I have updated the code, thanks again.